RHDEVDOCS-3307 Resource quota for pipelines (#38478)
openshift/openshift-docs: modules/op-alternative-approaches-compute-resource-quota-pipelines.adoc

//
// */cicd/pipelines/setting-compute-resource-quota-for-openshift-pipelines.adoc

[id="alternative-approaches-compute-resource-quota-pipelines_{context}"]
= Alternative approaches for limiting compute resource consumption in OpenShift Pipelines

To attain some degree of control over the usage of compute resources by a pipeline, consider the following alternative approaches:

* Set resource requests and limits for each step in a task.
+
.Example: Set resource requests and limits for each step in a task.
+ 
[source,yaml]
----
...
spec:
  steps:
    - name: step-with-limts
      resources:
        requests:
          memory: 1Gi
          cpu: 500m
        limits:
          memory: 2Gi
          cpu: 800m
...
----

* Set resource limits by specifying values for the `LimitRange` object. For more information on `LimitRange`, refer to xref:../../nodes/clusters/nodes-cluster-limit-ranges.adoc[Restrict resource consumption with limit ranges].

* xref:../../cicd/pipelines/reducing-pipelines-resource-consumption.adoc[Reduce pipeline resource consumption].

* Set and manage xref:../../applications/quotas/quotas-setting-per-project.adoc[resource quotas per project].

> * Ideally, the compute resource quota for a pipeline should be same as the total amount of compute resources consumed by the concurrently running pods in a pipeline run. However, the pods running the tasks consume compute resources based on the use case. For example, a Maven build task might require different amount of compute resources for different application that are built. As a result, you cannot predetermine the compute resource quotas for tasks in a generic pipeline. For greater predictability and control over usage of compute resources, use customized pipelines for different applications.

Reviewer: ```suggestion
* Ideally, the compute resource quota for a pipeline should be same as the total amount of compute resources consumed by the concurrently running pods in a pipeline run. However, the pods running the tasks consume compute resources based on the use case. For example, a Maven build task might require different compute resources for different applications that it builds. As a result, you cannot predetermine the compute resource quotas for tasks in a generic pipeline. For greater predictability and control over usage of compute resources, use customized pipelines for different applications.
```
